docs/architecture.md

```diff
@@ -2,7 +2,38 @@
 
 This doc covers the architectural design of Kubeless and directory structure of the repository.
 
-## Architecture
+## Concepts
+
+Kubeless is built around below core concepts:
+
+- Functions
+- Triggers
+- Runtime
+
+### Functions
+
+Function is representation of the code to be executed, with specification and metadata like runtime dependencies, build instructions etc. A function has a independent life-cycle and allows below method to control it:
```

Comment:
I would redact this a bit different. Just as a suggestion:

A _Function_ is a representation of the code to be executed. Along with the code, a _Function_ contains metadata about its runtime dependencies, build instructions, etc. A _Function_ has an independent life-cycle. The following methods are supported: